### DISCUSSION TITLE
How to append dynamic children in Tree view

### DISCUSSION THREAD

**Discussion by jithureddy** [airbnb/visx#1644] [General]
Hi,

Need to understand how to add dynamic children data after the initial tree data is loaded on to the page

Like Collapsible tree in here #162 where the data is always loaded upfront like this https://codesandbox.io/s/n3w687vmqj?file=/data.js

I want to load the data when child nodes are clicked, want to understand how to update the Tree dynamically when new data is loaded.

**Comment by williaster** [comment]
Hi @jithureddy 👋  I'm going to convert this to a discussion since it's a usage question. To load data dynamically you should just update the full hierarchy on node click. (I'm not aware of any other approach, you could search dynamically updating `d3` hierarchies as well since the same solution would apply here. the main difference would be you don't have to think about `d3`'s enter/update/exit complexities since that's what react will do for you).

The potential complexity would come from the layout for the old nodes __changing__ positions. I'm not sure there's a way to avoid this, especially if you are constricted to the same `width/height` for your overall tree the node positions would likely have to change to accommodate more nodes. Hope that helps.
